### PR TITLE
MODFIN-65- Adding currency during FY creation

### PR DESCRIPTION
**Pull request by KVupp**
## Purpose
https://issues.folio.org/browse/MODFIN-65

## Approach
- fetch config from mod-configuration
- if the config is empty use the default values, values fetched from ui-tenant-settings. 
The locale entry is created only when the settings are edited
- Also set the RMB to latest version

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/helper/FiscalYearsHelper.java [hunk 1]
```diff
@@ -28,8 +33,15 @@ public FiscalYearsHelper(HttpClientInterface httpClient, Map<String, String> oka
...
   }
 
-  public CompletableFuture<FiscalYear> createFiscalYear(FiscalYear fund) {
-    return handleCreateRequest(resourcesPath(FISCAL_YEARS), fund).thenApply(fund::withId);
+  public CompletableFuture<FiscalYear> createFiscalYear(FiscalYear fiscalYear) {
+    return HelperUtils.getConfigurationEntries(okapiHeaders, ctx, logger)
+      .thenCompose(locale -> {
+        // Initially the config will not have any locale, the same default values used here are hardcoded in tenant Settings
+        fiscalYear.setCurrency(new JsonObject(
+            locale.getString(LOCALE_SETTINGS, DEFAULT_LOCALE))
```
Comment (pautau-s) [inline src/main/java/org/folio/rest/helper/FiscalYearsHelper.java]: It is theoretically possible that the configuration will have `LOCALE_SETTINGS`, but there will be no currency in it.

Comment (KVupp) [reply src/main/java/org/folio/rest/helper/FiscalYearsHelper.java]: added additional tests

File: src/main/java/org/folio/rest/util/ErrorCodes.java [hunk 2]
```diff
@@ -9,7 +9,8 @@
   FISCAL_YEARS_NOT_FOUND("fiscalYearsNotFound", "Cannot find current fiscal year for specified ledger"),
   GROUP_NOT_FOUND("groupNotFound", "Cannot find group"),
   INVALID_TRANSACTION_TYPE("invalidTransactionType", "Invalid transaction type"),
-  INVALID_TRANSACTION_COUNT("invalidTransactionCount", "Number of transactions have to be greater than or equal to 1");
+  INVALID_TRANSACTION_COUNT("invalidTransactionCount", "Number of transactions have to be greater than or equal to 1"),
+  CURRENCY_NOT_FOUND("currencyNotFound", "Failed to fetch curreny from Locale");
```
Comment (javalkar-varun) [inline src/main/java/org/folio/rest/util/ErrorCodes.java]: Typo `curreny` -> `currency`

File: src/test/java/org/folio/rest/impl/FiscalYearTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  @Test
+  public void testPostFiscalYearErrorConfigurationEntry() {
+    logger.info("=== Test create FiscalYear with Error ===");
+    Headers headers = prepareHeaders(X_OKAPI_URL, ERROR_X_OKAPI_TENANT, X_OKAPI_TOKEN);
+    verifyPostResponse(FISCAL_YEAR.getEndpoint(), FISCAL_YEAR.getMockObject(), headers, APPLICATION_JSON,
+        INTERNAL_SERVER_ERROR.getStatusCode()).as(Errors.class);
+  }
+
+  @Test
+  public void testPutFiscalYearWithCurreny() {
```
Comment (javalkar-varun) [inline src/test/java/org/folio/rest/impl/FiscalYearTest.java]: Typo `Curreny`

File: src/test/java/org/folio/rest/impl/FiscalYearTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import org.folio.rest.jaxrs.model.FiscalYear;
+import org.folio.rest.util.ErrorCodes;
+import org.folio.rest.util.MockServer;
+import org.junit.jupiter.api.Test;
+
+public class FiscalYearTest extends ApiTestBase {
+  private static final Logger logger = LoggerFactory.getLogger(FiscalYearTest.class);
+
+  @Test
+  public void testPostFiscalYearWithCurreny() {
```
Comment (javalkar-varun) [inline src/test/java/org/folio/rest/impl/FiscalYearTest.java]: typo `Curreny`

Comment (KVupp) [reply src/test/java/org/folio/rest/impl/FiscalYearTest.java]: fixed

File: src/test/java/org/folio/rest/impl/FiscalYearTest.java (hunk 3 above)
Comment (KVupp) [reply src/test/java/org/folio/rest/impl/FiscalYearTest.java]: fixed

File: src/main/java/org/folio/rest/util/ErrorCodes.java (hunk 2 above)
Comment (KVupp) [reply src/main/java/org/folio/rest/util/ErrorCodes.java]: done

File: src/test/java/org/folio/rest/impl/FiscalYearTest.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    Headers headers = prepareHeaders(X_OKAPI_URL, EMPTY_CONFIG_X_OKAPI_TENANT, X_OKAPI_TOKEN);
+    JsonObject body = FISCAL_YEAR.getMockObject();
+
+    verifyPut(FISCAL_YEAR.getEndpointWithId((String) body.remove(ID)), body.toString(), headers, "", NO_CONTENT.getStatusCode());
+
+    List<JsonObject> rqRsPostFund = MockServer.getRqRsEntries(HttpMethod.PUT, FISCAL_YEAR.name());
+    assertThat(rqRsPostFund.get(0)
+      .getString("currency"), notNullValue());
+    assertEquals(rqRsPostFund.get(0)
```
Comment (javalkar-varun) [inline src/test/java/org/folio/rest/impl/FiscalYearTest.java]: Sonar cloud detected code smell. Seems like the actual and expected arguments should be swapped.

**Pull request merged**